client/src/components/EventDetail/EventDetail.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import EventDatesTable from "../EventDatesTable/EventDatesTable";
+import CurrentWeather from "../CurrentWeather/CurrentWeather";
+import { useParams } from "react-router";
+import { getEvent } from "./EventDetailFunctions";
+import { PollingEvent } from "../../types";
+
+function useFetchData(id: string | undefined): { data: PollingEvent | undefined; loading: boolean; error: Error | undefined } {
```

Comment:
Tady bych asi zmenil, ze `id` je povinny parametr (smazat to `| undefined`) a pak na radku 34 to "vynutit"
```
const { data, loading, error } = useFetchData(id as string);
```

realne to asi bude vzdycky string. Jinak by jsi mel asi v teto metode osetrit, aby pro undefined se nesnazila nic delat. Coz (vzhledem k pouziti s routerem) je asi trosku zbytecna prace navic.